### PR TITLE
Export ops.cond

### PR DESCRIPTION
**Pull request by AmedeoBiolatti**
Export `keras_core.ops.cond`

### CODE REVIEW

File: keras_core/ops/core.py [hunk 1]
```diff
@@ -374,3 +375,18 @@ def convert_to_numpy(x):
...
     return backend.convert_to_numpy(x)
+
+
+@keras_core_export("keras_core.ops.cond")
+def cond(pred, true_fn, false_fn):
+    """Conditionally applies `true_fn` or `false_fn`.
+
+    Args:
+        pred: Boolean scalar type
+        true_fn: A callable
```
Comment (fchollet) [inline keras_core/ops/core.py:386]: "Callable returning the output for the `pred == True` case."
""Callable returning the output for the `pred == False` case."

File: keras_core/ops/core_test.py [hunk 2]
```diff
@@ -247,3 +247,9 @@ def test_convert_to_tensor(self):
 
         with self.assertRaises(ValueError):
             ops.convert_to_numpy(KerasTensor((2,)))
+
+    def test_cond(self):
```
Comment (fchollet) [inline keras_core/ops/core_test.py:251]: Please also add a symbolic shape inference test case.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/ops/core_test.py (hunk 2 above)
Comment (AmedeoBiolatti) [reply keras_core/ops/core_test.py:251]: To make the symbolic shape inference possible in all cases I had to implement a Cond operation to assert the `true_fn` and `false_fn` shapes to be equal. Hope that this isn't against the design philosophy

File: keras_core/ops/core.py [hunk 3]
```diff
@@ -374,3 +375,35 @@ def convert_to_numpy(x):
...
+        self.true_fn = true_fn
+        self.false_fn = false_fn
+
+    def call(self, pred):
+        return backend.core.cond(pred, self.true_fn, self.false_fn)
+
+    def compute_output_spec(self, pred):
+        true_output_spec = backend.compute_output_spec(self.true_fn)
+        false_output_spec = backend.compute_output_spec(self.false_fn)
+        assert true_output_spec.dtype == false_output_spec.dtype
```
Comment (fchollet) [inline keras_core/ops/core.py:392]: Prefer using a `ValueError` that will print the different output shapes/dtypes -- that will be a better UX.

**Comment by AmedeoBiolatti** [comment]
Making `compute_output_spec` work with lambdas without breaking anything seems to be more complex than expected. I cannot fully undestand the error with the torch backend at the moment

**Comment by fchollet** [comment]
If we can't make it work, an option is to make `cond` eager-only, i.e. make it error out with a nice message when passed symbolic tensors. Then there is no new logic needed.

What's the error with the torch backend?

**Comment by fchollet** [comment]
> an option is to make cond eager-only

On second thought, this is likely the best move. The benefits of being able to build a Functional API model with conditional nodes are unclear

**Comment by AmedeoBiolatti** [comment]
> If we can't make it work, an option is to make `cond` eager-only, i.e. make it error out with a nice message when passed symbolic tensors. Then there is no new logic needed.
> 
> What's the error with the torch backend?

I solved the torch error with the last commit, now everything works fine for both eager and symbolic mode! The error was in a unit test for a lr schedule that use cond. It was something in the way I overwritten `__call__` for the `Cond` operator, but I have rewritten this part of the code and now it works fine

**Review by fchollet — APPROVED** [review]
LGTM -- thank you for the contribution!

**Pull request merged**